src/Controllers/FilterController.cs

```diff
@@ -84,16 +84,20 @@ public IActionResult SubjectWizardPost(ResultsFilter filter)
         private IActionResult FullTextPost(ResultsFilter filter)
         {
             filter = filter.WithoutLocation();
+            Boolean queryIsEmpty = string.IsNullOrWhiteSpace(filter.query);
+            List<Provider> suggestions = _api.GetProviderSuggestions(filter.query);
+            Boolean noSuggestions = suggestions.Count == 0;
+            Boolean queryIsExactMatch = suggestions.Any(x => string.Compare(filter.query, x.Name, CultureInfo.InvariantCulture, CompareOptions.IgnoreCase) == 0);
 
-            if (string.IsNullOrWhiteSpace(filter.query))
+            if (queryIsEmpty || noSuggestions)
             {
                 TempData.Put("Errors", new ErrorViewModel("query", "Training provider", "Please enter the name of a training provider", Url.Action("Location")));
                 return RedirectToAction("Location", filter.ToRouteValues());
             }
-            else if (false == _api.GetProviderSuggestions(filter.query).Any(x => string.Compare(filter.query, x.Name, CultureInfo.InvariantCulture, CompareOptions.IgnoreCase) == 0))
+            else if (!queryIsExactMatch)
             {
-                TempData.Put("Errors", new ErrorViewModel("query", "Training provider", "Please enter the name of a training provider", Url.Action("Location")));
-                return RedirectToAction("Location", filter.ToRouteValues());
+                TempData.Put("Suggestions", suggestions);
```

Comment:
ie move it here

Reply:
We can't because it's used for another check above; I've refactored the code slightly.